content/articles/how-to-handle-file-uploads-with-flask/index.md

```diff
@@ -0,0 +1,255 @@
+
+```
+mkdir tutorial
+```
+NOTE this is my project name choice yours can be anything.
+To build a flask application we must install flask, to do that we will use the below command 
+```
+pip install flask 
+
+```
+After installation, we will create a new file with the name ```app.py```, update the file with the below code
+
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return"hello world"
+
+if __name__==('__main__'):
+    app.run(debug=True)
+```
+from the code above we are importing flask from the flask library we installed.
+
+The ```@app.route``` is doing the routing for us, index is our view function which will return our page content to the browser.
+
+The if statement return the ```app.run``` which will enable us to run our app and refresh our page whenever we save changes.
+
+to run our app we run the below command on our terminal
+```python
+python app.py
+```
+note that ```app.py``` is the name of my app yours can be different. if everything goes well you will have a result like the one shown below.
+
+[image1]
+
+To upload files we will use the WTforms and the flask-uploads library. To work with this library we need to install them. do that with the code below
+```
+pip install flask_wtf, WTForms
+```
+```
+pip install flask-uploads
+```
+after installation, we will create a file field, by updating the code to the one below 
+```python 
+from flask import Flask, render_template
+from flask_wtf import FlaskForm
+from wtforms import FileField
+
+app = Flask(__name__)
+
+class MyForm(FlaskForm):
+    image = FileField('image')
+
+@app.route('/')
+def index():
+    form = MyForm()
+    return render_template('index.html')
+
+if __name__==('__main__'):
+    app.run(debug=True)
+```
+from the code above we start by importing FlaskForm from flask_wtf and FileField from wtforms. Then we created a class for our form as ```Myform``` image is the file field our image files will be saved to, then we call our Form class in our index function. We changed our render to render template, this is also a flask library used for rendering HTML templates from the code we rendered ```index.html```. Now let us create the HTML template we are rendering. update the code to the one below to do that 
+```python
+!doctype html>
+<html>
+  <head>
+    <title>File Upload</title>
+  </head>
+  <body>
+
+    <form action="/" method="POST" enctype="multipart/form-data">
+        {{ form.csrf_token }}
+        {{ form.image }}
+        <button type="submit">upload</button>
+    </form>
+  </body>
+</html>
+```
+From the above code, our form takes a method ```POST``` because we will be posting a file the ```csrf_token``` is a built-in function that handles security for us, then we call our form field we created in our ```Form Class``` using ```form.image```
+
+now we can run our app using ```python app.py``` if everything is correct you will get a runtime error like shown in the below image.
```

Comment:
```suggestion
From the above code, our form takes a method `POST` because we will be posting a file the `csrf_token` is a built-in function that handles security for us, then we call our form field we created in our `Form Class` using `form.image`. Now we can run our app using `python app.py`. If everything is correct, you will get a runtime error like shown in the below image.
```